thorlcr/master/thdemonserver.cpp

```diff
@@ -93,7 +93,12 @@ class DeMonServer : public CSimpleInterface, implements IDeMonServer
             const unsigned clusterWidth = queryNodeClusterWidth();
             const cost_type sgCost = money2cost_type(calcCost(thorManagerRate, duration) + calcCost(thorWorkerRate, duration) * clusterWidth);
             if (finished && sgCost)
+            {
                 wu->setStatistic(queryStatisticsComponentType(), queryStatisticsComponentName(), SSTsubgraph, graphScope, StCostExecute, NULL, sgCost, 1, 0, StatsMergeReplace);
+                cost_type costDiskAccess = graph.getDiskAccessCost();
```

Comment:
I think it's doable.  @ghalliday what's your view this suggestion?

Reply:
@ghalliday @shamser - I think my feeling is still that the CMasterGraph::getDiskAccessCost() code shouldn't be needed,
and instead the existing stats are aggregated in same way that eclagent does using the generic aggregateDiskAccessCost helper.

Reply:
makes sense, I was also worried that it would be less efficient if used a general mechanism that needed to pull back stats from Dali - I wasn't sure how significant though.
And yes, it was the extra code - or rather non-common code that performs aggregation in different ways

Given there's a separate JIRA already to look at general aggregation, and given efficiency, I think it's fine as is.